### PR TITLE
Remove deprecated getVariant method

### LINKED ISSUE #22380: Remove deprecated ResolvedComponentResult#getVariant method

### PR DESCRIPTION
**Pull request by tresat**
Fixes #22380

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

File: subprojects/architecture-test/src/changes/accepted-public-api-changes.json [hunk 1]
```diff
@@ -96,6 +96,14 @@
                 "Method has been removed"
             ]
         },
+        {
+            "type": "org.gradle.api.artifacts.result.ResolvedComponentResult",
+            "member": "Method org.gradle.api.artifacts.result.ResolvedComponentResult.getVariant()",
+            "acceptation": "[ADD YOUR CUSTOM REASON HERE]",
```
Comment (big-guy) [inline subprojects/architecture-test/src/changes/accepted-public-api-changes.json:102]: need a reason?

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**